app/src/main/java/org/piwigo/helper/URLHelper.java

```diff
@@ -0,0 +1,94 @@
+package org.piwigo.helper;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.piwigo.ui.login.LoginActivity;
+import org.piwigo.ui.main.MainActivity;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.UnknownHostException;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLPeerUnverifiedException;
+
+public class URLHelper extends AsyncTask<String, Void, String> {
+
+    private AsyncUrlResponse delegate;
+
+    public URLHelper(AsyncUrlResponse delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    protected String doInBackground(String... url) {
+        String newUrl = url[0].replaceAll("https://", "").replaceAll("http://", "");
+        try {
+            if (isHttpsWebsite(newUrl))
+                return ("https://" + newUrl);
+            else if (isHttpWebsite(newUrl))
+                return ("http://" + newUrl);
+            else
+                return ("https://" + newUrl);
+        } catch (IOException e) {
+            Log.e("URLHelper", "IOException", e.getCause());
+            return ("https://" + newUrl);
+        }
```

Comment:
In case of the IOException, the requested URL used would have HTTPS as prefix, but this exception SHOULD happen only whether no Piwigo is installed / the URL is not valid.

Should we inform the user of the exception, then ?

Reply:
I think we should, yes. Today we get a "Login failed, please try again." (not even translated) but this is confusing in case there is web server responding on http or https, but no piwigo installation is found at given location. Especially in case it is due to a typo in the site address, the user would more search for a typo in the user name and password...

This might be a little unrelated to the IOException, but I think it is worth to try to understand from the docs WHEN exactly the IOException is fired and output an appropriate message. - Maybe it could happen if the network connection is lost after the host was resolved or something that happens rarely. Still we should give the user info that a plain retry without changing site address or credentials could help.

Reply:
Another question: Why do you log only e.getCause() and not just e?

Reply:
This is what Android Studio recommended in arguments, as Log.e ask for a Throwable as last arg.. :)

Reply:
But "e" itself is already a Throwable... the "getCause" seems only relevant in case of chained exceptions